service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseRoleService.java

```diff
@@ -27,9 +34,11 @@
 
     private final IdamClient idam;
     private final CaseUserApi caseUser;
+    private final CaseAccessDataStoreApi caseAccessDataStoreApi;
     private final AuthTokenGenerator authTokenGenerator;
     private final SystemUpdateUserConfiguration userConfig;
     private final OrganisationService organisationService;
+    private final AddCaseAssignedUserRolesRequest addCaseAssignedUserRolesRequest;
```

Comment:
It will be injected by spring as a singleton, which is wrong. Each time fresh object should be created in a method when needed